src/main/java/net/harawata/appdirs/AppDirsFactory.java

```diff
@@ -25,16 +25,32 @@ private AppDirsFactory() {
     super();
   }
 
+  /**
+   * @return platform dependent implementation of <code>AppDirs</code>. Since
+   *         1.4.0, the same instance is returned for repeated invocations.
+   */
   public static AppDirs getInstance() {
-    String os = System.getProperty("os.name").toLowerCase();
-    if (os.startsWith("mac os x")) {
-      return new MacOSXAppDirs();
-    } else if (os.startsWith("windows")) {
-      WindowsFolderResolver folderResolver = new ShellFolderResolver();
-      return new WindowsAppDirs(folderResolver);
-    } else {
-      // Assume other *nix.
-      return new UnixAppDirs();
+    return Holder.INSTANCE;
+  }
+
+  /** Singleton instance holder. */
+  private static class Holder {
```

Comment:
Nitpick: singletons should be done with `enum`: https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#3-enforce-the-singleton-property-with-a-private-constructor-or-an-enum-type

Reply:
I am not sure if we can apply the enum singleton here.
If you have a clear idea, could you show me the essential part?